### ISSUE TITLE
CVE-2023-51766 - Exim (through 4.97) vulnerable to SMTP smuggling attack

### ISSUE THREAD

**Issue by rawhub-io** [MailCleaner/MailCleaner#571]
Good evening,

are you already aware of the zeroday-ish SMTP smuggling attack presented at Chaos Communication Congress 37C3 last week.

This also affects Exim through 4.97 and seems to be patched with 4.97.1.

German CERT-Bund rates this issue with a CVSS Base Score of 8.3 (high) and a CVSS Temporal Score of 7.5 (high) ([Link](https://wid.cert-bund.de/portal/wid/securityadvisory?name=WID-SEC-2023-3206))

Please see here for more information:
https://github.com/advisories/GHSA-w6ww-869j-cgm6

As well as the original blog post detailing the attack:
https://sec-consult.com/blog/detail/smtp-smuggling-spoofing-e-mails-worldwide/

Recording of the talk/presentation:
https://media.ccc.de/v/37c3-11782-smtp_smuggling_spoofing_e-mails_worldwide

**Comment by JohnMertz** [comment]
Hi,

I had seen some activity on the Exim mailing list, but did not have the time to take a look until now, since it did not appear to being discussed with the same gravity as other recent vulnerabilities. 

It appears that Exim in MailCleaner is only vulnerable to the outbound attack (ie. producing the second message from an authenticated sender within the same transaction). So we have a case where MailCleaner users could potentially spoof other users of the same MailCleaner cluster (or other domains that include the MailCleaner IP in their SPF).

As I currently understand it, the GitHub advisory is not exactly correct for the outgoing attack. The vulnerability does not allow for bypassing of SPF since the spoofed sender must necessarily have the an IP in common with the actual sender. SPF is simply a mechanism to confirm the alignment of the sending domain and the source IP, so this is seen as a 'pass' and it should be. What it does allow for is spoofing a message from a different account that shares an SPF IP by piggybacking on a different set of valid credentials. The first step in the attack mentioned in the blog post is key: "First, we register e-mail accounts at various public e-mail providers". In our case, that means opening an account or a domain which is hosted by a MailCleaner cluster (which actually has authenticated relaying on). So the possibility of exploiting this to spoof an account on that cluster is mostly limited to our ISPs and resellers, bad-actors within an organization, or compromised accounts.

From an inbound filtering perspective, this isn't really any different than a run-of-the-mill From Header spoof where 'my_company.com' is a google business customer:

```
> HELO ME
> MAIL FROM: <malicious_actor@gmail.com>
> RCPT TO: <credulous_recipient@my_company.com>
> DATA
FROM: "Your Boss" <boss@my_company.com>
To: "Credulous Recipient" <credulous_recipient@my_company.com>
...
```

The place this will presumably be exploited most will be specialty written programs sending these malformed messages without use of a hosted mail service, since they will all have been patched quickly.

In that case, however, it will be easier to discover even than the run-of-the-mill case because it will always show up as two message sent back-to-back from different Envelope Sender addresses in the same transaction. Presumably this could even be stopped by setting "Maxium messages per connection" to 1 (although that would slow down bulk-mailers also).

Once patched, it also does not prevent these messages from being submitted incoming, it just means that it will remain as one message and that message will have the "correct" From Header, if it is not also spoofed.

This is not to say that I have no regard for the possible headaches it could cause for administrators. I'm simply noting that from MailCleaner's point of view, it does not inhibit the all of the other tools that would be used to detect all of the spam heuristics still included within the attack message.

It is a severe enough problem, especially for our larger institutional and ISP users that we should provide a new release.

That said, I had a quick look at our standard build process for a new package and it failed to build 4.97.1. It appears that there are some function dependency changes which may be causing problems with our old versions of OpenSSL. Unfortunately newer versions of OpenSSL create further problems with our old version of MySQL.

I'm going to fight with the dependencies a bit longer, but if it's not easy to reconcile, I will probably just port the relevant changes and push a patched version of 4.96.

A bright note for the future is that [my docker env build script](https://github.com/MailCleaner/mc-exim) is able to build it without a problem for our [upcoming Bookworm release](https://github.com/MailCleaner/MailCleaner-Next/issues).

**Comment by JohnMertz** [comment]
After extensive testing to accommodate the new, longer Exim ID's, I have just released exim 4.97.1 which resolves this issue.
